### PR TITLE
Usb work for IMXRT

### PR DESCRIPTION
**Pull request by HarrisonChoi750405**
Add usb ehci driver fuction.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [b8040c268cc4ee5f034e1e2ff6040b4728645f49] - Code Rule Check (C++) OK.

File: os/arch/arm/src/imxrt/chip/imxrt_daisy.h [hunk 1]
```diff
@@ -75,4 +75,5 @@
 
 void imxrt_daisy_select(unsigned int index, unsigned int alt);
 
-#endif							/* __ARCH_ARM_SRC_IMXRT_CHIP_IMXRT_DAISY_H */
+#endif  /* __ARCH_ARM_SRC_IMXRT_CHIP_IMXRT_DAISY_H */
+
```
Comment (drashti304) [inline os/arch/arm/src/imxrt/chip/imxrt_daisy.h]: Remove extra line

File: os/arch/arm/src/imxrt/chip/imxrt_lpi2c.h [hunk 2]
```diff
@@ -220,4 +220,5 @@
 #define IMXRT_LPI2C4_STDR                   (IMXRT_LPI2C4_BASE + IMXRT_LPI2C_STDR_OFFSET)	/* Slave Transmit Data Register  */
 #define IMXRT_LPI2C4_SRDR                   (IMXRT_LPI2C4_BASE + IMXRT_LPI2C_SRDR_OFFSET)	/* Slave Receive Data Register  */
 
+
```
Comment (drashti304) [inline os/arch/arm/src/imxrt/chip/imxrt_lpi2c.h]: Remove the extra line

File: os/arch/arm/src/imxrt/imxrt_userspace.c [hunk 3]
```diff
@@ -94,8 +94,8 @@ void imxrt_userspace(void)
 
 	DEBUGASSERT(USERSPACE->us_bssstart != 0 && USERSPACE->us_bssend != 0 && USERSPACE->us_bssstart <= USERSPACE->us_bssend);
 
-	dest = (uint8_t *)USERSPACE->us_bssstart;
-	end = (uint8_t *)USERSPACE->us_bssend;
+	dest = (uint8_t *) USERSPACE->us_bssstart;
```
Comment (yashwanth686007) [inline os/arch/arm/src/imxrt/imxrt_userspace.c]: Please revert & correct coding guideline violations.

**Comment by sangwon03** [comment]
Please resolve conflicts

**Comment by seinfra** [comment]
Target : [9d5a061950b1a36feabf32438b76a2bc8b93fed5] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9d5a061950b1a36feabf32438b76a2bc8b93fed5] - Code Rule Check Result:
os/arch/arm/src/imxrt/chip/imxrt105x_config.h:1313: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
os/arch/arm/src/imxrt/chip/imxrt105x_config.h:35015: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
os/arch/arm/src/imxrt/chip/imxrt105x_config.h:44113: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
os/arch/arm/src/imxrt/imxrt_ehci.c:132: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_ehci.c:169: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/imxrt_ehci.c:169: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_ehci.c:170: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/arm/src/imxrt/imxrt_ehci.c:170: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_ehci.c:196: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/imxrt/imxrt_ehci.c:196: ERROR: [SPC_M_OPR] space required after that ',' (ctx:VxV)
os/arch/arm/src/imxrt/imxrt_ehci.c:266: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/imxrt/imxrt_ehci.c:267: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/imxrt/imxrt_ehci.c:268: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/imxrt/imxrt_ehci.c:269: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/imxrt/imxrt_ehci.c:275: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/imxrt/imxrt_ehci.c:276: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/arm/src/imxrt/imxrt_ehci.c:277: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/board/imxrt1050-evk/src/imxrt_bringup.c:420: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/imxrt1050-evk/src/imxrt_bringup.c:421: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/imxrt1050-evk/src/imxrt_bringup.c:422: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/imxrt1050-evk/src/imxrt_bringup.c:422: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/board/imxrt1050-evk/src/imxrt_bringup.c:423: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/imxrt1050-evk/src/imxrt_bringup.c:424: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/imxrt1050-evk/src/imxrt_bringup.c:425: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/imxrt1050-evk/src/imxrt_bringup.c:426: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Target : [c3fdafb61f9eaf2414850d3582079432d08777c2] -

**Comment by seinfra** [comment]
Target : [c3fdafb61f9eaf2414850d3582079432d08777c2] -

**Comment by seinfra** [comment]
Target : [299a567d99838c652dd74c693ec8fa3dfdec028b] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0f631af28abfe8f66e0e23bcf50a519435ba8c4c] - Code Rule Check Result:
os/arch/arm/src/imxrt/imxrt_lowputc.c:251: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/arch/arm/src/imxrt/imxrt_lowputc.c:252: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/esp32_devkit/src/esp32_boot.c:228: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/include/debug.h:158: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/debug.h:204: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/include/debug.h:247: ERROR: [IDT_M_TAB] code indent should use tabs where possible

**Comment by seinfra** [comment]
Target : [0f631af28abfe8f66e0e23bcf50a519435ba8c4c] - Code Rule Check (C++) OK.

File: os/arch/arm/src/imxrt/imxrt_ehci.h [hunk 4]
```diff
@@ -0,0 +1,108 @@
+/************************************************************************************
+ * arch/arm/src/imxrt/imxrt_ehci.h
```
Comment (NilabhShant) [inline os/arch/arm/src/imxrt/imxrt_ehci.h]: 1. correct the path:
arch/arm/src/imxrt/imxrt_ehci.h  --> os/arch/arm/src/imxrt/imxrt_ehci.h

2. Add Samsung License part as well.

File: os/arch/arm/src/imxrt/imxrt_usbhost.c [hunk 5]
```diff
@@ -0,0 +1,343 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/************************************************************************************
+ * arch/arm/src/imxrt_usbhost.c
```
Comment (NilabhShant) [inline os/arch/arm/src/imxrt/imxrt_usbhost.c]: same as above

**Comment by HarrisonChoi750405** [comment]
I am sorry
I have to discard this PR.
Because PR about iMXRT1020-EVK is commit and additional change is necessary.
I will submit another PR about it

**Pull request closed without merging**